## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 474901`
Commit: `997ecdea56d6dd129fe55a1991942d0f57c6038f`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 2 packages marked as broken and skipped:</summary>
  <ul>
    <li>netbox_4_2</li>
    <li>netbox_4_3</li>
  </ul>
</details>
<details>
  <summary>:x: 400 packages failed to build:</summary>
  <ul>
    <li>barbicanclient (python313Packages.python-barbicanclient)</li>
    <li>barbicanclient.dist (python313Packages.python-barbicanclient.dist)</li>
    <li>bashate</li>
    <li>bashate.dist</li>
    <li>breezy (python313Packages.breezy)</li>
    <li>breezy.dist (python313Packages.breezy.dist)</li>
    <li>cabal2nix</li>
    <li>calendar-cli</li>
    <li>calendar-cli.dist</li>
    <li>devpi-client</li>
    <li>devpi-client.dist</li>
    <li>dvc (python313Packages.dvc)</li>
    <li>dvc-with-remotes</li>
    <li>dvc-with-remotes.dist</li>
    <li>dvc.dist (python313Packages.dvc.dist)</li>
    <li>errands</li>
    <li>errbot</li>
    <li>errbot.dist</li>
    <li>flatpak-builder</li>
    <li>flatpak-builder.doc</li>
    <li>flatpak-builder.installedTests</li>
    <li>flatpak-builder.man</li>
    <li>gaphor</li>
    <li>gaphor.dist</li>
    <li>git-dumper</li>
    <li>git-dumper.dist</li>
    <li>glanceclient (python313Packages.python-glanceclient)</li>
    <li>glanceclient.dist (python313Packages.python-glanceclient.dist)</li>
    <li>gtg</li>
    <li>heatclient (python313Packages.python-heatclient)</li>
    <li>heatclient.dist (python313Packages.python-heatclient.dist)</li>
    <li>home-assistant-component-tests.caldav</li>
    <li>html5validator</li>
    <li>html5validator.dist</li>
    <li>hydra</li>
    <li>hydra.doc</li>
    <li>ironicclient (python313Packages.python-ironicclient)</li>
    <li>ironicclient.dist (python313Packages.python-ironicclient.dist)</li>
    <li>jenkins-job-builder (python313Packages.jenkins-job-builder)</li>
    <li>jenkins-job-builder.dist (python313Packages.jenkins-job-builder.dist)</li>
    <li>klaus (python313Packages.klaus)</li>
    <li>klaus.dist (python313Packages.klaus.dist)</li>
    <li>kolla</li>
    <li>kolla.dist</li>
    <li>luarocks-packages-updater</li>
    <li>luarocks-packages-updater.dist</li>
    <li>magnumclient (python313Packages.python-magnumclient)</li>
    <li>magnumclient.dist (python313Packages.python-magnumclient.dist)</li>
    <li>manilaclient (python313Packages.python-manilaclient)</li>
    <li>manilaclient.dist (python313Packages.python-manilaclient.dist)</li>
    <li>mistralclient (python313Packages.python-mistralclient)</li>
    <li>mistralclient.dist (python313Packages.python-mistralclient.dist)</li>
    <li>netbox (netbox_4_4)</li>
    <li>neutronclient (python313Packages.python-neutronclient)</li>
    <li>neutronclient.dist (python313Packages.python-neutronclient.dist)</li>
    <li>nix-prefetch-bzr</li>
    <li>nix-prefetch-scripts</li>
    <li>nix-update-source</li>
    <li>nix-update-source.dist</li>
    <li>open-web-calendar</li>
    <li>open-web-calendar.dist</li>
    <li>openstackclient (python313Packages.python-openstackclient)</li>
    <li>openstackclient-full</li>
    <li>openstackclient-full.dist</li>
    <li>openstackclient.dist (python313Packages.python-openstackclient.dist)</li>
    <li>peering-manager</li>
    <li>poetry</li>
    <li>poetry.dist</li>
    <li>poetryPlugins.poetry-audit-plugin</li>
    <li>poetryPlugins.poetry-audit-plugin.dist</li>
    <li>poetryPlugins.poetry-plugin-export</li>
    <li>poetryPlugins.poetry-plugin-export.dist</li>
    <li>poetryPlugins.poetry-plugin-shell</li>
    <li>poetryPlugins.poetry-plugin-shell.dist</li>
    <li>poetryPlugins.poetry-plugin-up</li>
    <li>poetryPlugins.poetry-plugin-up.dist</li>
    <li>prowler</li>
    <li>prowler.dist</li>
    <li>python312Packages.breezy</li>
    <li>python312Packages.breezy.dist</li>
    <li>python312Packages.caldav</li>
    <li>python312Packages.caldav.dist</li>
    <li>python312Packages.check-manifest</li>
    <li>python312Packages.check-manifest.dist</li>
    <li>python312Packages.cliff</li>
    <li>python312Packages.cliff.dist</li>
    <li>python312Packages.comet-ml</li>
    <li>python312Packages.comet-ml.dist</li>
    <li>python312Packages.cometx</li>
    <li>python312Packages.cometx.dist</li>
    <li>python312Packages.debtcollector</li>
    <li>python312Packages.debtcollector.dist</li>
    <li>python312Packages.dulwich</li>
    <li>python312Packages.dulwich.dist</li>
    <li>python312Packages.dvc</li>
    <li>python312Packages.dvc-hdfs</li>
    <li>python312Packages.dvc-hdfs.dist</li>
    <li>python312Packages.dvc-studio-client</li>
    <li>python312Packages.dvc-studio-client.dist</li>
    <li>python312Packages.dvc.dist</li>
    <li>python312Packages.dvclive</li>
    <li>python312Packages.dvclive.dist</li>
    <li>python312Packages.executorch</li>
    <li>python312Packages.executorch.dist</li>
    <li>python312Packages.gto</li>
    <li>python312Packages.gto.dist</li>
    <li>python312Packages.hacking</li>
    <li>python312Packages.hacking.dist</li>
    <li>python312Packages.hg-git</li>
    <li>python312Packages.hg-git.dist</li>
    <li>python312Packages.jenkins-job-builder</li>
    <li>python312Packages.jenkins-job-builder.dist</li>
    <li>python312Packages.keystoneauth1</li>
    <li>python312Packages.keystoneauth1.dist</li>
    <li>python312Packages.klaus</li>
    <li>python312Packages.klaus.dist</li>
    <li>python312Packages.mmcv</li>
    <li>python312Packages.mmcv.dist</li>
    <li>python312Packages.mmengine</li>
    <li>python312Packages.mmengine.dist</li>
    <li>python312Packages.netbox-bgp</li>
    <li>python312Packages.netbox-bgp.dist</li>
    <li>python312Packages.netbox-documents</li>
    <li>python312Packages.netbox-documents.dist</li>
    <li>python312Packages.netbox-interface-synchronization</li>
    <li>python312Packages.netbox-interface-synchronization.dist</li>
    <li>python312Packages.netbox-plugin-prometheus-sd</li>
    <li>python312Packages.netbox-plugin-prometheus-sd.dist</li>
    <li>python312Packages.netbox-qrcode</li>
    <li>python312Packages.netbox-qrcode.dist</li>
    <li>python312Packages.netbox-reorder-rack</li>
    <li>python312Packages.netbox-reorder-rack.dist</li>
    <li>python312Packages.netbox-routing</li>
    <li>python312Packages.netbox-routing.dist</li>
    <li>python312Packages.openstackdocstheme</li>
    <li>python312Packages.openstackdocstheme.dist</li>
    <li>python312Packages.openstacksdk</li>
    <li>python312Packages.openstacksdk.dist</li>
    <li>python312Packages.openstacksdk.man</li>
    <li>python312Packages.os-client-config</li>
    <li>python312Packages.os-client-config.dist</li>
    <li>python312Packages.osc-lib</li>
    <li>python312Packages.osc-lib.dist</li>
    <li>python312Packages.osc-placement</li>
    <li>python312Packages.osc-placement.dist</li>
    <li>python312Packages.oslo-concurrency</li>
    <li>python312Packages.oslo-concurrency.dist</li>
    <li>python312Packages.oslo-config</li>
    <li>python312Packages.oslo-config.dist</li>
    <li>python312Packages.oslo-context</li>
    <li>python312Packages.oslo-context.dist</li>
    <li>python312Packages.oslo-db</li>
    <li>python312Packages.oslo-db.dist</li>
    <li>python312Packages.oslo-i18n</li>
    <li>python312Packages.oslo-i18n.dist</li>
    <li>python312Packages.oslo-log</li>
    <li>python312Packages.oslo-log.dist</li>
    <li>python312Packages.oslo-metrics</li>
    <li>python312Packages.oslo-metrics.dist</li>
    <li>python312Packages.oslo-serialization</li>
    <li>python312Packages.oslo-serialization.dist</li>
    <li>python312Packages.oslo-utils</li>
    <li>python312Packages.oslo-utils.dist</li>
    <li>python312Packages.osprofiler</li>
    <li>python312Packages.osprofiler.dist</li>
    <li>python312Packages.pygerber</li>
    <li>python312Packages.pygerber.dist</li>
    <li>python312Packages.pyghmi</li>
    <li>python312Packages.pyghmi.dist</li>
    <li>python312Packages.python-aodhclient</li>
    <li>python312Packages.python-aodhclient.dist</li>
    <li>python312Packages.python-barbicanclient</li>
    <li>python312Packages.python-barbicanclient.dist</li>
    <li>python312Packages.python-cinderclient</li>
    <li>python312Packages.python-cinderclient.dist</li>
    <li>python312Packages.python-designateclient</li>
    <li>python312Packages.python-designateclient.dist</li>
    <li>python312Packages.python-glanceclient</li>
    <li>python312Packages.python-glanceclient.dist</li>
    <li>python312Packages.python-heatclient</li>
    <li>python312Packages.python-heatclient.dist</li>
    <li>python312Packages.python-ironicclient</li>
    <li>python312Packages.python-ironicclient.dist</li>
    <li>python312Packages.python-jenkins</li>
    <li>python312Packages.python-jenkins.dist</li>
    <li>python312Packages.python-keystoneclient</li>
    <li>python312Packages.python-keystoneclient.dist</li>
    <li>python312Packages.python-magnumclient</li>
    <li>python312Packages.python-magnumclient.dist</li>
    <li>python312Packages.python-manilaclient</li>
    <li>python312Packages.python-manilaclient.dist</li>
    <li>python312Packages.python-mistralclient</li>
    <li>python312Packages.python-mistralclient.dist</li>
    <li>python312Packages.python-neutronclient</li>
    <li>python312Packages.python-neutronclient.dist</li>
    <li>python312Packages.python-novaclient</li>
    <li>python312Packages.python-novaclient.dist</li>
    <li>python312Packages.python-octaviaclient</li>
    <li>python312Packages.python-octaviaclient.dist</li>
    <li>python312Packages.python-openstackclient</li>
    <li>python312Packages.python-openstackclient.dist</li>
    <li>python312Packages.python-swiftclient</li>
    <li>python312Packages.python-swiftclient.dist</li>
    <li>python312Packages.python-troveclient</li>
    <li>python312Packages.python-troveclient.dist</li>
    <li>python312Packages.python-watcherclient</li>
    <li>python312Packages.python-watcherclient.dist</li>
    <li>python312Packages.python-zaqarclient</li>
    <li>python312Packages.python-zaqarclient.dist</li>
    <li>python312Packages.python-zunclient</li>
    <li>python312Packages.python-zunclient.dist</li>
    <li>python312Packages.scmrepo</li>
    <li>python312Packages.scmrepo.dist</li>
    <li>python312Packages.simplekv</li>
    <li>python312Packages.simplekv.dist</li>
    <li>python312Packages.stestr</li>
    <li>python312Packages.stestr.dist</li>
    <li>python312Packages.swh-auth</li>
    <li>python312Packages.swh-auth.dist</li>
    <li>python312Packages.swh-core</li>
    <li>python312Packages.swh-core.dist</li>
    <li>python312Packages.swh-export</li>
    <li>python312Packages.swh-export.dist</li>
    <li>python312Packages.swh-journal</li>
    <li>python312Packages.swh-journal.dist</li>
    <li>python312Packages.swh-model</li>
    <li>python312Packages.swh-model.dist</li>
    <li>python312Packages.swh-objstorage</li>
    <li>python312Packages.swh-objstorage.dist</li>
    <li>python312Packages.swh-scanner</li>
    <li>python312Packages.swh-scanner.dist</li>
    <li>python312Packages.swh-scheduler</li>
    <li>python312Packages.swh-scheduler.dist</li>
    <li>python312Packages.swh-storage</li>
    <li>python312Packages.swh-storage.dist</li>
    <li>python312Packages.swh-web-client</li>
    <li>python312Packages.swh-web-client.dist</li>
    <li>python312Packages.tempest</li>
    <li>python312Packages.tempest.dist</li>
    <li>python312Packages.torchtune</li>
    <li>python312Packages.torchtune.dist</li>
    <li>python313Packages.caldav</li>
    <li>python313Packages.caldav.dist</li>
    <li>python313Packages.check-manifest</li>
    <li>python313Packages.check-manifest.dist</li>
    <li>python313Packages.cliff</li>
    <li>python313Packages.cliff.dist</li>
    <li>python313Packages.comet-ml</li>
    <li>python313Packages.comet-ml.dist</li>
    <li>python313Packages.cometx</li>
    <li>python313Packages.cometx.dist</li>
    <li>python313Packages.debtcollector</li>
    <li>python313Packages.debtcollector.dist</li>
    <li>python313Packages.dulwich</li>
    <li>python313Packages.dulwich.dist</li>
    <li>python313Packages.dvc-hdfs</li>
    <li>python313Packages.dvc-hdfs.dist</li>
    <li>python313Packages.dvc-studio-client</li>
    <li>python313Packages.dvc-studio-client.dist</li>
    <li>python313Packages.dvclive</li>
    <li>python313Packages.dvclive.dist</li>
    <li>python313Packages.executorch</li>
    <li>python313Packages.executorch.dist</li>
    <li>python313Packages.gto</li>
    <li>python313Packages.gto.dist</li>
    <li>python313Packages.hacking</li>
    <li>python313Packages.hacking.dist</li>
    <li>python313Packages.hg-git</li>
    <li>python313Packages.hg-git.dist</li>
    <li>python313Packages.keystoneauth1</li>
    <li>python313Packages.keystoneauth1.dist</li>
    <li>python313Packages.mmcv</li>
    <li>python313Packages.mmcv.dist</li>
    <li>python313Packages.mmengine</li>
    <li>python313Packages.mmengine.dist</li>
    <li>python313Packages.netbox-attachments</li>
    <li>python313Packages.netbox-attachments.dist</li>
    <li>python313Packages.netbox-bgp</li>
    <li>python313Packages.netbox-bgp.dist</li>
    <li>python313Packages.netbox-contract</li>
    <li>python313Packages.netbox-contract.dist</li>
    <li>python313Packages.netbox-documents</li>
    <li>python313Packages.netbox-documents.dist</li>
    <li>python313Packages.netbox-floorplan-plugin</li>
    <li>python313Packages.netbox-floorplan-plugin.dist</li>
    <li>python313Packages.netbox-interface-synchronization</li>
    <li>python313Packages.netbox-interface-synchronization.dist</li>
    <li>python313Packages.netbox-napalm-plugin</li>
    <li>python313Packages.netbox-napalm-plugin.dist</li>
    <li>python313Packages.netbox-plugin-prometheus-sd</li>
    <li>python313Packages.netbox-plugin-prometheus-sd.dist</li>
    <li>python313Packages.netbox-qrcode</li>
    <li>python313Packages.netbox-qrcode.dist</li>
    <li>python313Packages.netbox-reorder-rack</li>
    <li>python313Packages.netbox-reorder-rack.dist</li>
    <li>python313Packages.netbox-routing</li>
    <li>python313Packages.netbox-routing.dist</li>
    <li>python313Packages.netbox-topology-views</li>
    <li>python313Packages.netbox-topology-views.dist</li>
    <li>python313Packages.openstackdocstheme</li>
    <li>python313Packages.openstackdocstheme.dist</li>
    <li>python313Packages.openstacksdk</li>
    <li>python313Packages.openstacksdk.dist</li>
    <li>python313Packages.openstacksdk.man</li>
    <li>python313Packages.os-client-config</li>
    <li>python313Packages.os-client-config.dist</li>
    <li>python313Packages.osc-lib</li>
    <li>python313Packages.osc-lib.dist</li>
    <li>python313Packages.osc-placement</li>
    <li>python313Packages.osc-placement.dist</li>
    <li>python313Packages.oslo-concurrency</li>
    <li>python313Packages.oslo-concurrency.dist</li>
    <li>python313Packages.oslo-config</li>
    <li>python313Packages.oslo-config.dist</li>
    <li>python313Packages.oslo-context</li>
    <li>python313Packages.oslo-context.dist</li>
    <li>python313Packages.oslo-db</li>
    <li>python313Packages.oslo-db.dist</li>
    <li>python313Packages.oslo-i18n</li>
    <li>python313Packages.oslo-i18n.dist</li>
    <li>python313Packages.oslo-log</li>
    <li>python313Packages.oslo-log.dist</li>
    <li>python313Packages.oslo-metrics</li>
    <li>python313Packages.oslo-metrics.dist</li>
    <li>python313Packages.oslo-serialization</li>
    <li>python313Packages.oslo-serialization.dist</li>
    <li>python313Packages.oslo-utils</li>
    <li>python313Packages.oslo-utils.dist</li>
    <li>python313Packages.osprofiler</li>
    <li>python313Packages.osprofiler.dist</li>
    <li>python313Packages.pygerber</li>
    <li>python313Packages.pygerber.dist</li>
    <li>python313Packages.pyghmi</li>
    <li>python313Packages.pyghmi.dist</li>
    <li>python313Packages.python-aodhclient</li>
    <li>python313Packages.python-aodhclient.dist</li>
    <li>python313Packages.python-cinderclient</li>
    <li>python313Packages.python-cinderclient.dist</li>
    <li>python313Packages.python-designateclient</li>
    <li>python313Packages.python-designateclient.dist</li>
    <li>python313Packages.python-jenkins</li>
    <li>python313Packages.python-jenkins.dist</li>
    <li>python313Packages.python-keystoneclient</li>
    <li>python313Packages.python-keystoneclient.dist</li>
    <li>python313Packages.python-novaclient</li>
    <li>python313Packages.python-novaclient.dist</li>
    <li>python313Packages.python-octaviaclient</li>
    <li>python313Packages.python-octaviaclient.dist</li>
    <li>swiftclient (python313Packages.python-swiftclient)</li>
    <li>swiftclient.dist (python313Packages.python-swiftclient.dist)</li>
    <li>troveclient (python313Packages.python-troveclient)</li>
    <li>troveclient.dist (python313Packages.python-troveclient.dist)</li>
    <li>watcherclient (python313Packages.python-watcherclient)</li>
    <li>watcherclient.dist (python313Packages.python-watcherclient.dist)</li>
    <li>python313Packages.python-zaqarclient</li>
    <li>python313Packages.python-zaqarclient.dist</li>
    <li>zunclient (python313Packages.python-zunclient)</li>
    <li>zunclient.dist (python313Packages.python-zunclient.dist)</li>
    <li>python313Packages.scmrepo</li>
    <li>python313Packages.scmrepo.dist</li>
    <li>python313Packages.simplekv</li>
    <li>python313Packages.simplekv.dist</li>
    <li>python313Packages.stestr</li>
    <li>python313Packages.stestr.dist</li>
    <li>python313Packages.swh-auth</li>
    <li>python313Packages.swh-auth.dist</li>
    <li>python313Packages.swh-core</li>
    <li>python313Packages.swh-core.dist</li>
    <li>python313Packages.swh-export</li>
    <li>python313Packages.swh-export.dist</li>
    <li>python313Packages.swh-journal</li>
    <li>python313Packages.swh-journal.dist</li>
    <li>python313Packages.swh-model</li>
    <li>python313Packages.swh-model.dist</li>
    <li>python313Packages.swh-objstorage</li>
    <li>python313Packages.swh-objstorage.dist</li>
    <li>python313Packages.swh-scanner</li>
    <li>python313Packages.swh-scanner.dist</li>
    <li>python313Packages.swh-scheduler</li>
    <li>python313Packages.swh-scheduler.dist</li>
    <li>python313Packages.swh-storage</li>
    <li>python313Packages.swh-storage.dist</li>
    <li>python313Packages.swh-web-client</li>
    <li>python313Packages.swh-web-client.dist</li>
    <li>python313Packages.tempest</li>
    <li>python313Packages.tempest.dist</li>
    <li>python313Packages.torchtune</li>
    <li>python313Packages.torchtune.dist</li>
    <li>reno</li>
    <li>reno.dist</li>
    <li>sigil</li>
    <li>silver-platter</li>
    <li>silver-platter.dist</li>
    <li>swh</li>
    <li>vcstool</li>
    <li>vcstool.dist</li>
    <li>watchgha</li>
    <li>watchgha.dist</li>
    <li>xandikos</li>
    <li>xandikos.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>python312Packages.dulwich</summary>
<pre>  /build/source/tests/test_walk.py:37: PytestCollectionWarning: cannot collect test class &#x27;TestWalkEntry&#x27; because it has a __init__ constructor (from: tests/test_walk.py)
    class TestWalkEntry:

tests/test_web.py:83
  /build/source/tests/test_web.py:83: PytestCollectionWarning: cannot collect test class &#x27;TestHTTPGitRequest&#x27; because it has a __init__ constructor (from: tests/test_web.py)
    class TestHTTPGitRequest(HTTPGitRequest):

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/test_repository.py::SharedRepositoryTests::test_existing_repo_respects_config - PermissionError: [Errno 1] Operation not permitted: &#x27;/build/tmprd6vwjkm/bra...
FAILED tests/test_repository.py::SharedRepositoryTests::test_index_file_permissions_group - PermissionError: [Errno 1] Operation not permitted: &#x27;/build/tmpjksw_paw/.gi...
FAILED tests/test_repository.py::SharedRepositoryTests::test_init_bare_shared_all - PermissionError: [Errno 1] Operation not permitted: &#x27;/build/tmp_6yltt40/bra...
FAILED tests/test_repository.py::SharedRepositoryTests::test_init_bare_shared_group - PermissionError: [Errno 1] Operation not permitted: &#x27;/build/tmpvcyjqe_e/bra...
FAILED tests/test_repository.py::SharedRepositoryTests::test_loose_object_permissions_all - PermissionError: [Errno 1] Operation not permitted: &#x27;/build/tmp4r9rv3lu/bra...
FAILED tests/test_repository.py::SharedRepositoryTests::test_loose_object_permissions_group - PermissionError: [Errno 1] Operation not permitted: &#x27;/build/tmp17yeew0l/bra...
FAILED tests/test_repository.py::SharedRepositoryTests::test_pack_file_permissions_group - PermissionError: [Errno 1] Operation not permitted: &#x27;/build/tmpfj7gl0pp/bra...
FAILED tests/test_repository.py::SharedRepositoryTests::test_pack_index_permissions_group - PermissionError: [Errno 1] Operation not permitted: &#x27;/build/tmpxawni0n8/bra...
FAILED tests/test_repository.py::SharedRepositoryTests::test_reflog_permissions_group - PermissionError: [Errno 1] Operation not permitted: &#x27;/build/tmpsojn5dwu/.gi...
FAILED tests/test_sha256_pack.py::SHA256PackTests::test_pack_index_v1_with_sha256 - TypeError: pack index v1 only supports SHA-1 names
= 10 failed, 3003 passed, 60 skipped, 18 deselected, 1 xfailed, 11 warnings in 144.51s (0:02:24) =</pre>
</details>
<details>
<summary>python313Packages.dulwich</summary>
<pre>  /build/source/tests/test_walk.py:37: PytestCollectionWarning: cannot collect test class &#x27;TestWalkEntry&#x27; because it has a __init__ constructor (from: tests/test_walk.py)
    class TestWalkEntry:

tests/test_web.py:83
  /build/source/tests/test_web.py:83: PytestCollectionWarning: cannot collect test class &#x27;TestHTTPGitRequest&#x27; because it has a __init__ constructor (from: tests/test_web.py)
    class TestHTTPGitRequest(HTTPGitRequest):

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/test_repository.py::SharedRepositoryTests::test_existing_repo_respects_config - PermissionError: [Errno 1] Operation not permitted: &#x27;/build/tmpaurf2w48/bra...
FAILED tests/test_repository.py::SharedRepositoryTests::test_index_file_permissions_group - PermissionError: [Errno 1] Operation not permitted: &#x27;/build/tmp8zdvafr_/.gi...
FAILED tests/test_repository.py::SharedRepositoryTests::test_init_bare_shared_all - PermissionError: [Errno 1] Operation not permitted: &#x27;/build/tmp5ttioldu/bra...
FAILED tests/test_repository.py::SharedRepositoryTests::test_init_bare_shared_group - PermissionError: [Errno 1] Operation not permitted: &#x27;/build/tmp9fzr_ix3/bra...
FAILED tests/test_repository.py::SharedRepositoryTests::test_loose_object_permissions_all - PermissionError: [Errno 1] Operation not permitted: &#x27;/build/tmpitsvoedl/bra...
FAILED tests/test_repository.py::SharedRepositoryTests::test_loose_object_permissions_group - PermissionError: [Errno 1] Operation not permitted: &#x27;/build/tmp3e8rc_ke/bra...
FAILED tests/test_repository.py::SharedRepositoryTests::test_pack_file_permissions_group - PermissionError: [Errno 1] Operation not permitted: &#x27;/build/tmp1op85kyx/bra...
FAILED tests/test_repository.py::SharedRepositoryTests::test_pack_index_permissions_group - PermissionError: [Errno 1] Operation not permitted: &#x27;/build/tmpjjqj_a61/bra...
FAILED tests/test_repository.py::SharedRepositoryTests::test_reflog_permissions_group - PermissionError: [Errno 1] Operation not permitted: &#x27;/build/tmp0cx8z7qj/.gi...
FAILED tests/test_sha256_pack.py::SHA256PackTests::test_pack_index_v1_with_sha256 - TypeError: pack index v1 only supports SHA-1 names
= 10 failed, 3003 passed, 60 skipped, 18 deselected, 1 xfailed, 11 warnings in 144.06s (0:02:24) =</pre>
</details>
</details>

--- a/pkgs/development/python-modules/dulwich/default.nix
+++ b/pkgs/development/python-modules/dulwich/default.nix
@@ -110,6 +110,9 @@ buildPythonPackage rec {
       asl20
       gpl2Plus
     ];
-    maintainers = with lib.maintainers; [ koral ];
+    maintainers = with lib.maintainers; [
+      koral
+      sarahec
+    ];
   };
 }

--- a/pkgs/development/python-modules/dulwich/default.nix
+++ b/pkgs/development/python-modules/dulwich/default.nix
@@ -13,13 +13,11 @@
   merge3,
   paramiko,
   pytestCheckHook,
-  pythonOlder,
   rich,
   rustPlatform,
   rustc,
   setuptools,
   setuptools-rust,
-  typing-extensions,
   urllib3,
 }:
 
@@ -27,8 +25,6 @@ buildPythonPackage rec {
   pname = "dulwich";
   version = "0.24.1";
   pyproject = true;
-
-  disabled = pythonOlder "3.9";
 
   src = fetchFromGitHub {
     owner = "jelmer";
@@ -55,9 +51,6 @@ buildPythonPackage rec {
 
   dependencies = [
     urllib3
-  ]
-  ++ lib.optionals (pythonOlder "3.11") [
-    typing-extensions
   ];
 
   optional-dependencies = {

--- a/pkgs/development/python-modules/dulwich/default.nix
+++ b/pkgs/development/python-modules/dulwich/default.nix
@@ -81,17 +81,17 @@ buildPythonPackage rec {
     "test_file_win"
     # dulwich.errors.NotGitRepository: No git repository was found at .
     "WorktreeCliTests"
-    # 'SwiftPackData' object has no attribute '_file'
-    "test_iterobjects_subset_all_present"
-    "test_iterobjects_subset_missing_allowed"
-    "test_iterobjects_subset_missing_not_allowed"
     # Adding a symlink to a directory outside the repo doesn't raise
     "test_add_symlink_absolute_to_system"
+    # Depends on setuid which is not available in sandboxed environments
+    "SharedRepositoryTests"
+    # TypeError: pack index v1 only supports SHA-1 names
+    "test_pack_index_v1_with_sha256"
   ];
 
   disabledTestPaths = [
-    # requires swift config file
-    "tests/contrib/test_swift_smoke.py"
+    # "Code [in contrib] is not an official part of Dulwich, and may no longer work"
+    "tests/contrib"
   ];
 
   __darwinAllowLocalNetworking = true;

